src/math.rs

```diff
@@ -225,17 +264,22 @@ impl OrientedBeam {
 
     pub fn contains(&self, p: &Point3<f64>) -> bool {
         // What is the point in beam coordinates?
-        let Point3 { x, y, .. } = self.rotation_inv.rotate_point(*p) + self.translation_inv;
+        let Point3 { x, y, .. } =
+            self.isometry_inv.rotation.rotate_point(*p) + self.isometry_inv.translation;
         x.abs() <= self.half_extent.x && y.abs() <= self.half_extent.y
     }
 
+    pub fn transform(&self, isometry: &Isometry3<f64>) -> Self {
+        Self::new(isometry * &self.isometry_inv.inverse(), self.half_extent)
+    }
+
     fn precompute_corners(
-        rotation: &Quaternion<f64>,
-        translation: &Vector3<f64>,
+        isometry: &Isometry3<f64>,
         half_extent: &Vector2<f64>,
     ) -> [Point3<f64>; 4] {
-        let transform =
-            |x: f64, y: f64| rotation.rotate_point(Point3::new(x, y, 0.0)) + translation;
+        let transform = |x: f64, y: f64| {
+            isometry.rotation.rotate_point(Point3::new(x, y, 0.0)) + isometry.translation
+        };
         [
             transform(half_extent.x, half_extent.y),
```

Comment:
could be this transform be confused with the function in line 272? could it be more explicit like get_corner_from_offset

Reply:
Yes, definitely, I renamed it.